compiler/noirc_evaluator/src/ssa/opt/remove_bit_shifts.rs

```diff
@@ -189,7 +189,7 @@ impl Context<'_, '_, '_> {
         }
 
         let to_bits = self.context.dfg.import_intrinsic(Intrinsic::ToBits(Endian::Little));
-        let max_exponent_bits = self.context.dfg.get_value_max_num_bits(exponent).ilog2() + 1;
+        let max_exponent_bits = self.context.dfg.type_of_value(exponent).bit_size().ilog2();
```

Comment:
Maybe we could add a comment here about the thought process with why `ilog2` gives us the exponents bit size. Would make it clearer for external contributors and auditors.

Reply:
Done! While documenting this I realized there's an optimization opportunity when the exponent was upcasted from `u1` so I handled that too. It's likely a very rare case, but why not :-)